src/naive/date.rs

```diff
@@ -1248,6 +1250,23 @@ impl NaiveDate {
         NaiveWeek { date: *self, start }
     }
 
+    // I think self here is a copy, but I still have to make sure it's
+    pub fn checked_add_years(self, years: Years) -> Option<NaiveDate> {
+        if years.0 == 0 {
+            return Some(self);
+        }
+        let num_years = i32::try_from(years.0).ok()?;
+        self.with_year(self.year() + num_years)
```

Comment:
This will fail if you add 1 year to February 29th. This is different from `checked_add_months`, which will fall back to the last day of the month.

Reply:
How do you handle integer overflow in for example `self.year() + `i32::MAX`?